### ISSUE TITLE
Document usage before 1.0 release

### ISSUE THREAD

**Issue by yurishkuro** [open-telemetry/opentelemetry-erlang#209]
This is a suggestion, but I feel pretty strongly about it. The current Usage section in the README only talks about how enable the library.  For the official 1.0 release there needs to be a section that describes the usage of the API for instrumentation, including sections on In-Process Context Propagation, Inter-process Propagation, and Tracing.

Cf. https://github.com/opentracing/opentracing-go#api-overview-for-those-adding-instrumentation

**Comment by tsloughter** [comment]
Yup, completely agreed. I've already started on the cross process propagation examples as part of pages for opentelemetry.io and will break it out from there once I have it ready and make a PR to the docs in this repo to go along with it.

Thanks for the Go reference, that'll be useful to compare with.

**Comment by yurishkuro** [comment]
NB: it's fine to point to OTEL.io site if it's going to have the same documentation, it's better to keep a single copy. With OpenTracing we usually kept the website to "representative examples" while the repo's README (or other generated docs) were the official usage.

**Comment by tsloughter** [comment]
Yea, I was thinking the otel.io site would focus more on examples while pointing to docs on hexdocs.pm that is generated by the project for deeper dives into the functioning of those examples.

So it is somewhat keeping multiple copies but a very slimmed down very on otel.io that is hopefully enough for anyone who doesn't have to do more than basic instrumentation covered by the macros.
